endive-gui/CodeMirror/src/main.ts

```diff
@@ -1,15 +1,86 @@
-import { EditorView, minimalSetup } from "codemirror";
-import { languageServer } from "codemirror-languageserver";
-import { placeholder } from "@codemirror/view";
-import { codeFolding, foldGutter } from "@codemirror/language";
+import {EditorView, minimalSetup} from "codemirror"
+import { languageServer } from 'codemirror-languageserver'
+import {placeholder} from "@codemirror/view"
+import {codeFolding, foldGutter} from "@codemirror/language"
 
-import { keymap } from "@codemirror/view";
-import { insertTab, indentLess } from "@codemirror/commands";
+import {keymap} from "@codemirror/view"
+import {insertTab,indentLess} from "@codemirror/commands"
+import {forEachDiagnostic} from "@codemirror/lint"
 
 import { foldNodeProp, foldInside } from "@codemirror/language";
 
-import { styleTags, tags } from "@lezer/highlight";
+import type { EditorState, Range } from '@codemirror/state'
+import { RangeSet, StateField } from '@codemirror/state'
+import type { DecorationSet } from '@codemirror/view'
+import { Decoration, WidgetType } from '@codemirror/view'
 
+
+interface SideLintParams {
+  content: string,
+  position: number
+}
+
+
+class SideLintWidget extends WidgetType {
+  readonly content
+  readonly position
+
+  constructor({ content, position }: SideLintParams) {
+    super()
+
+    this.content = content;
+    this.position = position;
+  }
+
+  eq(sideLint: SideLintWidget) {
+    return sideLint.content === this.content
+  }
+
+  toDOM() {
+    const container = document.createElement('span')
+    
+    container.setAttribute('aria-hidden', 'true')
+    container.classList.add("cm-sideLint")
+    let pos = this.position;
+    container.onclick = function(){
+      globalThis.editor.dispatch({selection: {anchor: pos, head: pos}})
+    };
+
+    container.innerHTML = "     /"+this.content
```

Comment:
This should be `textContent` because we don't want the browser to interpret HTML (do we?).